### DISCUSSION TITLE
Gantt Chart Functionality

### DISCUSSION THREAD

**Discussion by RoshanRajAyyathanKaruppal** [pegasystems/constellation-ui-gallery#51] [General]
Based on my understanding, it seems that two distinct data pages were likely used to render information on the user interface, with one of them being parameterized. Could you clarify how the data from these data pages is structured? Specifically, I'm curious about the mapping process, there's no indication in the code regarding which data originates from which page. Could you clarify this mapping process?

<img width="332" alt="Screenshot 2024-04-15 175252" src="https://github.com/pegasystems/constellation-ui-gallery/assets/165133705/1c018541-aa1e-40bf-b011-07a328a6d54e">

**Comment by vijayjangid** [comment]
Hello,

There's a single datapage available, as configured [here](https://github.com/pegasystems/constellation-ui-gallery/blob/master/src/components/Pega_Extensions_GanttChart/config.json#L29), which furnishes a flat list of "Generic" items with the properties listed [here](https://github.com/pegasystems/constellation-ui-gallery/blob/master/src/components/Pega_Extensions_GanttChart/config.json#L35-L71). These properties constitute the widget's configuration and facilitate the mapping of datapage response fields to the component in advance.

Further details on widget configuration can be found [here](https://github.com/pegasystems/constellation-ui-gallery/blob/master/src/components/Pega_Extensions_GanttChart/Docs.mdx).

Regarding the provided screenshot, if you've configured the widget as shown below:

```
categoryFieldName: 'Type',
parentFieldName: 'ProjectID',
dependenciesFieldName: 'Dependencies',
startDateFieldName: 'StartDate',
endDateFieldName: 'EndDate',
progressFieldName: 'Progress',
```

The component utilizes this mapping to construct the chart based on the actual field values. For example, if 'P-1001' is categorized as a Project and Task 'T-1001' has its ProjectID set as 'P-1001', the component displays the task as a child of the specified project.

Hope this explanation clears things up for you.

**Comment by ricmars** [comment]
@RoshanRajAyyathanKaruppal is Vijay's explanation answering your question - Do you think we need to improve the documentation on how to use this component? Thans
